trace/obfuscate.go

```diff
@@ -22,7 +64,69 @@ type obfuscationVisitor struct {
 
 var _ vis.DefaultSpecVisitor = (*obfuscationVisitor)(nil)
 
+// EnterData processes the given data and obfuscates all the primitive values with zero values, regardless of it's meta data.
 func (*obfuscationVisitor) EnterData(self interface{}, ctx vis.SpecVisitorContext, d *pb.Data) Cont {
+	return ObfuscateWithZeroValue(d)
+}
+
+type partialObfuscationVisitor struct {
+	vis.DefaultSpecVisitorImpl
+	obfuscationOptions *Obfuscator
+}
+
+var _ vis.DefaultSpecVisitor = (*partialObfuscationVisitor)(nil)
+
+// EnterData processes the given data and obfuscates sensitive information based on the provided obfuscation options.
+// It does 3 checks to determine if the data contains sensitive information:
+// 1. It checks if spec is an HTTP Authroization or Cookie data and obfuscates the value.
+// 2. It cheks if spec is an HTTP Header or Query Param data and obfuscates the value if the key is in the list of sensitive keys.
+// 3. It applies regex patterns to all obfuscate sensitive primitive string values.
+func (pov *partialObfuscationVisitor) EnterData(self interface{}, ctx vis.SpecVisitorContext, d *pb.Data) Cont {
```

Comment:
I am unclear why, if we are only operating on primitive values, we didn't use EnterPrimitive originally.  Worth a thought if we are trying to operate on the wrong object here when the Visitor will dispatch on only primitives for us.

Reply:
Is this for regex values matching specifically?